iceoryx_utils/source/posix_wrapper/file_lock.cpp

```diff
@@ -0,0 +1,300 @@
+// Copyright (c) 2021 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "iceoryx_utils/posix_wrapper/file_lock.hpp"
+#include "iceoryx_utils/cxx/smart_c.hpp"
+#include "iceoryx_utils/platform/fcntl.hpp"
+#include "iceoryx_utils/platform/platform_correction.hpp"
+#include "iceoryx_utils/platform/unistd.hpp"
+
+#include <sys/file.h>
+
+namespace iox
+{
+namespace posix
+{
+FileLock::FileLock() noexcept
+{
+    this->m_isInitialized = false;
```

Comment:
I agree. However, this was copy & pasted from UDS:
```
/// @brief default constructor. The result is an invalid UnixDomainSocket object which can be reassigned later by
/// using the
/// move constructor.
UnixDomainSocket() noexcept;
```

I'll remove it there, too.

Reply:
The `Semaphore` also has this c'tor. I'll check the others tomorrow.

Reply:
Turns out a lot of classes need to be touched. Hence, I started on a new branch: iox-#404-remove-param-less-ctor-because-of-creation-pattern @elfenpiff Could you finish this off?

Reply:
Moved to #613